spec/mongoid/collection_configurable_spec.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+class CollectionConfigurableCapped
+  include Mongoid::Document
+
+  store_in collection_options: {
+    capped: true,
+    size: 10000
+  }
+end
+
+class CollectionConfigurableUnknownOptions
+  include Mongoid::Document
+
+  store_in collection_options: {
+    i_am_not_an_option: true,
+    let_me: { raise: true }
+  }
+end
+
+describe Mongoid::CollectionConfigurable do
+  before(:each) do
+    [
+      CollectionConfigurableCapped,
+      CollectionConfigurableUnknownOptions
+    ].each do |klaz|
+      klaz.collection.drop
+    end
+  end
+
+  context 'when collection does not exist' do
+    context 'with an unknown collection options' do
+      let(:subject) do
+        CollectionConfigurableUnknownOptions
+      end
+
+      it 'creates the collection' do
+        subject.create_collection
+        expect(
+          subject.collection.database.list_collections(filter: { name: subject.collection_name.to_s })
+        ).not_to be_empty
+      end
+    end
+
+    context 'with supported options' do
+      let(:subject) do
+        CollectionConfigurableCapped
+      end
+
+      it 'creates the collection' do
+        subject.create_collection
+        expect(
+          subject.collection.database.list_collections(filter: { name: subject.collection_name.to_s })
+        ).not_to be_empty
+      end
+    end
+
+    context 'with unsupported options' do
+
+      let(:subject) do
+        CollectionConfigurableCapped
+      end
+
+      before do
+        expect_any_instance_of(Mongo::Collection)
+          .to receive(:create)
+          .with({ capped: true, size: 10000 })
+          .and_raise(Mongo::Error::OperationFailure)
+      end
+
+      it 'raises an error' do
```

Comment:
why does this raise an error?

Reply:
Because of the we the `expect_any_instance_of` above. It the driver raises an error, Mongoid should, too.
However, I can see why this test might seem unclear, I reworked it.